usecases/schema/class.go

```diff
@@ -204,17 +206,25 @@ func (h *Handler) UpdateClass(ctx context.Context, principal *models.Principal,
 			return err
 		}
 
-		if initial.ReplicationConfig.Factor != updated.ReplicationConfig.Factor {
-			return fmt.Errorf("updating replication factor is not supported yet")
+		initialRF := initial.ReplicationConfig.Factor
+		updatedRF := updated.ReplicationConfig.Factor
+
+		if initialRF != updatedRF {
+			shardingState, err = h.scaleOut.Scale(ctx, className,
+				h.metaReader.CopyShardingState(className).Config, initialRF, updatedRF)
```

Comment:
This can be eventually consistent. So if another follower is also updating the sharding state, depending on the scaleout result it *could* go wrong. For example you assume the sharding state1, start doing calculation how to scale out. At the same time the sharding state change due to another operation (or a different scaleout/scalein) and once we're done here we'll commit an erronous class update to raft.

I think here it really matter we get at least a strongly consistent view of the sharding state (we'd need to add a `Query` endpoint specifically for it). And also ensure there's no way we can run two of them in parallel.

Otherwise ideally maybe this is something that should only be ran on the leader ? This at least would avoid the EC issue. However the problem is that we can't push "scale out" or "scale in" operation in raft because they are dependent on side effect that raft can't assume (disk size to know where the replicas will go, etc...) so that means that follower nodes on applying the log entry might assume a different sharding state.